### ISSUE TITLE
Modal: change header prop to node

### ISSUE THREAD

**Issue by CloudPower97** [react-materialize/react-materialize#867]
Hey @alextrastero, what do you think if we use a `node` instead of a `string` as a `header` `prop` for the `Modal`?
This allow for greater flexibility, for example one can use a `span` or insert an `Icon` in the `Header` or anything really
Let me know what do you think, so I can open a __P.R__ in case :smile:

**Comment by alextrastero** [comment]
👍 great idea, `node` could also just be a string to remain backwards compatible?

**Comment by CloudPower97** [comment]
Not sure, but we can get away with it doing that:

```javascript
icon: PropTypes.oneOfType([PropTypes.node, PropTypes.string])
```

**Comment by alextrastero** [comment]
Looks good, go for it @CloudPower97 :)

**Comment by CloudPower97** [comment]
Hey @alextrastero, I'm going to close this issue as it has been resolved by #890 :100:
